### ISSUE TITLE
Wanted: improved type error message

### ISSUE THREAD

**Issue by stefjoosten** [AmpersandTarski/Ampersand#1017]
#### Version of ampersand that was used 
I used Ampersand-v3.17.3 [feature/Archimate3:b1a0984d1]

#### What I got
I compiled a project, RAP3, and got the following message:
<img width="858" alt="afbeelding" src="https://user-images.githubusercontent.com/275768/70602204-25e5aa80-1bf4-11ea-8cbf-fb95e8158742.png">
The problem is that the relation in the spreadsheet, `name[Role*RoleName]` is not declared. It took me a while to spot this because the type error claims ambiguity while there is none.

#### What I want instead
I would like the message to also point out that the declaration of `name[Role*RoleName]` might be missing.

#### Steps to reproduce
1. 
2.
3.
4.

**Comment by hanjoosten** [comment]
This is an interesting case. `name` is the overloaded name of multiple relations. The typechecker assumes that it knows all relations. This case suggests that your intent was that there was even another relation with the name `name`. 
I wonder if @sjcjoosten can help out here, since he is the expert in typechecking.

**Comment by stefjoosten** [comment]
I think it is merely that the possibility of a missing declaration should be incorporated in the error message, based on the fact that the relation to be checked differs from all the options in the ambibuity list...

**Comment by hanjoosten** [comment]
I understand. The problem is that at the point where the error message is generated, there is no knowledge at all of the missing relation. It just doesn't exist.

**Comment by sjcjoosten** [comment]
That suggests that the datatype chosen at that point might not be the right one.. I'm guessing it's an expression? If we want to allow full expressions in the first row of the excel table, then there is little to be done indeed, but I believe the first row should just be relation names (are there examples where a flip ~ is added?)

I think that not reusing expressions for things that aren't expressions would be the most elegant solution. Apart from that, I could also attempt to improve the disambiguation messages for simple cases where a type is unambiguously known.

**Comment by stefjoosten** [comment]
Indeed, technically the name on the first row can be an expression. However, the allowed expressions are limited to `r` and `r~` to avoid explaining to the user what happens if he inserts an atom into `r;s` or worse.

There is a good reason to want both `r` and `r~`. In practice a spreadsheet is often given and the user writes two lines above the table. One with relation names and one with concept names. To accommodate a larger category of excel-sheets, it is useful to allow the user to write `r~`.
